cypress/support/commands.ts

```diff
@@ -73,6 +74,33 @@ const visitInspector = () => {
   });
 };
 
+/**
+ * Allows you to visit the /viz/:id page and mock
+ * its SSR return
+ */
+const visitVizWithNextPageProps = (
+  data: Partial<GetSourceFileSsrQuery['getSourceFile']> & { id: string },
+) => {
+  cy.visit(`/viz/${data.id}?noSSR`, {
+    onBeforeLoad: (win) => {
+      let nextData: any;
+
+      Object.defineProperty(win, '__NEXT_DATA__', {
```

Comment:
This is a neat solution to the problem. 👍 "Abusing" getters & setters (and other fun features of JS/browsers) like this is my jam 🤣 

I was wondering about potential downsides though and something that comes to mind is that this still results in an incorrect HTML - right now we don't really utilize it that much so it works OK but, in the future, this might be problematic because the true response is 404 and it won't hold the "correct" HTML. We are also still rendering the component in question - just with empty/stub props. Which can easily break if we start relying on some new data etc.

I was wondering a bit about alternatives that would not suffer from those problems and I think I've come up with a neat solution - that wouldn't be hard to implement. It took me some time to come up with this but it seems like a **super** easy solution could be leveraged:
```js
cy.visit(`/viz/${data.id}?ssr=${JSON.stringify(data)}`)
```
And just use the content of the `ssr` param on the other side to render what we need. A nice trait about this is that it should be completely type-safe, without the need for any casting etc (apart from the fact that we are going to `JSON.parse` on the other side)